homeassistant/components/acaia/entity.py

```diff
@@ -0,0 +1,58 @@
+"""Base class for acaia entities."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyacaia_async.acaiascale import AcaiaScale
+
+from homeassistant.const import CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AcaiaCoordinator
+
+
+@dataclass(kw_only=True, frozen=True)
+class AcaiaEntityDescription(EntityDescription):
+    """Description for acaia entities."""
+
+    available_fn: Callable[[AcaiaScale], bool] = lambda scale: scale.connected
+
+
+@dataclass
+class AcaiaEntity(CoordinatorEntity[AcaiaCoordinator]):
+    """Common elements for all entities."""
+
+    _attr_has_entity_name = True
+    entity_description: AcaiaEntityDescription
+
+    def __init__(
+        self,
+        coordinator: AcaiaCoordinator,
+        entity_description: AcaiaEntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._scale = coordinator.scale
+        self._attr_unique_id = f"{self._scale.mac}_{entity_description.key}"
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._scale.mac)},
+            name=coordinator.config_entry.data.get(CONF_NAME),
```

Comment:
You know what I'm going to say about this ;)

Reply:
Actually this is already your recommendation. CONF_NAME is only set when the device is autodiscovered. When set up manually evaluates to `None`.